.github/workflows/detekt.yml

```diff
@@ -20,12 +18,19 @@ jobs:
         with:
           java-version: 11
           distribution: zulu
-      - uses: burrunan/gradle-cache-action@v1
+      - uses: gradle/gradle-build-action@v2
         with:
           gradle-version: wrapper
-          # additional files to calculate key for dependency cache
-          gradle-dependencies-cache-key: |
-            buildSrc/**/Versions.kt
-      # Workaround for bug in gradle-cache-action + detekt
-      - name: Run gradle command
-        run: ./gradlew detektAll
+          arguments: |
+            detektAll
+      - name: Copy SARIF reports into a single directory
+        if: ${{ always() }}
```

Comment:
Why do we need this `if`?

Reply:
Because if detekt gradle task fails, we still want to upload reports to display info about those failures. By default, github actions will skip all steps after a failed one.